Tests/DatadogTests/Datadog/URLSessionAutoInstrumentation/DDURLSessionDelegateTests.swift

```diff
@@ -176,4 +181,29 @@ class DDURLSessionDelegateTests: XCTestCase {
 
         try Datadog.deinitializeOrThrow()
     }
+
+    func testWhenDDURLSessionDelegateInits_itPicksCorrectInterceptor() throws {
+        // given
+        Datadog.initialize(
+            appContext: .mockAny(),
+            configuration: Datadog.Configuration.builderUsing(
+                clientToken: "abc-123",
+                environment: "tests"
+            )
+            .track(firstPartyHosts: ["example.com"])
+            .build()
+        )
+        XCTAssertNotNil(URLSessionAutoInstrumentation.instance)
```

Comment:
We don't need to initialize the entire SDK. This is enough:
```swift
// Given

URLSessionAutoInstrumentation.instance = URLSessionAutoInstrumentation(
   configuration: .mockAny(),
   dateProvider: SystemDateProvider()
)
defer { URLSessionAutoInstrumentation.instance = nil }
```

When initializing the full SDK like this we enable many unnecessary things:
* Logs / Traces / RUM upload workers (they start running),
* Logs / Traces / RUM storage `DispatchQueues`,
* `URLSession` swizzler is applied,

which will slow down the tests execution and may lead to more flakiness.

It's already ensured by other tests (`DatadogTests`) that the initialization leads to configuring `URLSessionAutoinstrumentation` stack properly.